### DISCUSSION TITLE
Filter M2M

### DISCUSSION THREAD

**Discussion by myzinsky** [directus/directus#3606] [Feature Requests]
I created a "tag" functionality using M2M. I have one tag table where people could define tags. And in my ordinary data table people could add tags in a M2M fashion. I have the requirement that one data element could have more than one tag. 

Now it would be nice if one could filter accoring to these tags. For example select all data elements that are tagged as "bug".

However Filtering M2M does not seam to work. Indeed it would be a nice feature.

**Comment by rijkvanzanten** [comment]
@myzinsky 

You should be able to use `/items/elements?filter[tags.name][_eq]=bug` to filter the parent set (elements) on the relational value (tag.name = bug) 🙂

**Reply by myzinsky** [reply]
I'm talking about the GUI:

<img width="289" alt="image" src="https://user-images.githubusercontent.com/1970554/104110914-a46fb880-52dc-11eb-9e65-0444dbf57692.png">

So it would be nice if we could filter there the tags...

**Reply by rijkvanzanten** [reply]
Are you sure `tags` there is a m2m? Relational fields are shown with the dropdown arrow that allow you to select related nested fields 🤔 (like `Antwordbild D`)

**Reply by myzinsky** [reply]
yes it is M2M:

<img width="851" alt="image" src="https://user-images.githubusercontent.com/1970554/104225509-c5ecb380-5446-11eb-9662-d446c9707838.png">

**Reply by myzinsky** [reply]
I double checked it "Antwort Bild X >" is a M2One relation. There filtering works, but for M2M it does not work. Any idea how to fix that?

**Reply by myzinsky** [reply]
If you expect the same thing as for M2O then this here is even a bug!

**Reply by myzinsky** [reply]
@rijkvanzanten can you confirm that this is a bug?

**Reply by rijkvanzanten** [reply]
Could very well be! Feel free to open an issue with ideally a sql dump so i can reproduce and fix :)

**Reply by cmllr** [reply]
I can also confirm that M2M filters are not working on the API/ permission side. Currently running v9.0.0-rc.34.

**Reply by myzinsky** [reply]
https://github.com/directus/directus/issues/3935

**Comment by ashx404** [comment]
Even with API, is it possible to filter the same way as "all"works. 
For ex: User and Languages collections, a user can speak multiple languages and I need to find a user who can speak  English and German.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
